Refactor hamburger menu semantics

peterblazejewicz opened the pull request:
This PR refactors semantics of hamburger menu to use more user friendly/compliant solution (no divs - yeah) - the same as used in:
https://github.com/twbs/bootstrap/blob/master/docs/_includes/nav/main.html#L4
https://github.com/google/web-starter-kit/blob/master/app/index.html#L41

- change markup to use native button
- change style to retain original design and features

The solution was tested - except of IE. The iOS simulator was used to hunt down and fix a bug (line-height needs to be reset for buttons):

![menu](https://cloud.githubusercontent.com/assets/14539/5984604/04aa53a0-a8d8-11e4-873c-5eae1ac836e2.gif)

One can now trigger menu via any pointing device/software. 

Thanks!

samccone commented on src/stylesheets/_header.scss:
```diff
@@ -108,9 +108,15 @@
         color: #fff;
-        display: block;
+        line-height: inherit;
+        background: transparent;
```
in scss we can do

```scss
background: transparent {
  image: none
}
```

thejameskyle replied:
but why would you want to?

samccone replied:
It is feature of the language

thejameskyle replied:
It's gross and confusing though

peterblazejewicz commented:
@samccone @thejameskyle 
refactored and squashed
I have no habit to use nested properties Sass features as I work with Less every-day - so nice tip BTW.
I've changed button markup  to shorter and leaner version I think.

Thanks guys!

jdaudier commented:
@peterblazejewicz I like how you're referencing other sources that have solved the same issue. Keep it up!
